src/messaging/ExchangeContext.cpp

```diff
@@ -689,5 +695,25 @@ void ExchangeContext::ExchangeSessionHolder::GrabExpiredSession(const SessionHan
     GrabUnchecked(session);
 }
 
+bool ExchangeContext::IsPeerActive()
+{
+    // We can never assume the peer is active if we are sending a group message
+    if (IsGroupExchangeContext())
+        return false;
+
+    // If sender is not initiator, it means the peer sent the first message.
+    // This means the peer is active if we are sending a message to the peer.
```

Comment:
This seems to be assuming some context around when this is called ("when we are sending a message to the peer") that is not really clearly documented in the public API.

But also, why is initiator special here?  The Matter message flows I am aware of basically all have two kinds of messages: a message that expects a response, and the last message in the exchange.  That situation is symmetric between initiator and responder....

Note, though, that in some error-handling cases the other side can send a message they think is the last one in an exchange and we will respond to it.  But that might not matter here.

In general, I am having a hard time following exactly what this method is meant to compute and when, and why this is the right computation.

Reply:
resolving comment since it is no longer applicable